sdk/azidentity/aad_identity_client.go

```diff
@@ -415,14 +415,15 @@ func (c *aadIdentityClient) authenticateAuthCode(ctx context.Context, tenantID s
 }
 
 // createAuthorizationCodeAuthRequest creates a request for an Access Token for authorization_code grant types.
-func (c *aadIdentityClient) createAuthorizationCodeAuthRequest(ctx context.Context, tenantID string, clientID string, authCode string, clientSecret string, redirectURI string, scopes []string) (*azcore.Request, error) {
+func (c *aadIdentityClient) createAuthorizationCodeAuthRequest(ctx context.Context, tenantID string, clientID string, authCode string, clientSecret string, scopes []string) (*azcore.Request, error) {
 	data := url.Values{}
 	data.Set(qpGrantType, "authorization_code")
 	data.Set(qpClientID, clientID)
 	if clientSecret != "" {
 		data.Set(qpClientSecret, clientSecret) // only for web apps
 	}
-	data.Set(qpRedirectURI, redirectURI)
+	// similar to MSAL, we use a hard-coded redirect URI here as we never actually redirect to it
+	data.Set(qpRedirectURI, "https://login.microsoftonline.com/common/oauth2/nativeclient")
```

Comment:
I'm just curious if you know about any app type that might need a custom redirect URI? Or if the MSAL team ever mentioned any? Not blocking on it, but just curious about this.

Reply:
In our [conversation](https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/68#discussion_r517352378) the MSAL folks said that "MSAL stops the redirect" so there's no need to actually perform the redirect.

If we do end up needing this we can probably add it to the options struct given that it's optional from an MSAL perspective.

Reply:
Ok that sounds like a good plan to me.